Source/WebKit/UIProcess/AuxiliaryProcessProxy.h

```diff
@@ -78,6 +78,12 @@ class AuxiliaryProcessProxy
     using ResponsivenessTimer::Client::WeakValueType;
     using ResponsivenessTimer::Client::WeakPtrImplType;
 
+    // ProcessLauncher::Client
+    uint32_t ptrCount() const final { return IPC::Connection::Client::ptrCount(); }
+    uint32_t ptrCountWithoutThreadCheck() const final { return IPC::Connection::Client::ptrCountWithoutThreadCheck(); }
+    void incrementPtrCount() const final { IPC::Connection::Client::incrementPtrCount(); }
+    void decrementPtrCount() const final { IPC::Connection::Client::decrementPtrCount(); }
```

Comment:
Can this be achieved with `using` instead?

Reply:
The alternative would be to have `ProcessLauncher::Client` subclass `CanMakeCheckedPtr`, and have `AuxiliaryProcessProxy` use `using` statements to disambiguate since it would inherit 2 checked ptr counts from 2 different base classes.

I can go this way if you'd prefer but I thought it was nicer to avoid inheriting 2 different checked ptr counts (cc @geoffreygaren)

Reply:
I think we want to avoid multiple checked counts. It's an unclear programming model.

My goal is to get to the point where every class defines its lifetime semantics in exactly one place, in a concrete class. 

Side point: IPC::Connection::Client should probably make its CheckedPtr interface virtual, just like ProcessLauncher::Client. In my view, it's an anti-pattern when an abstract base class mandates a concrete lifetime implementation. 

Another side point: RefPtr+WeakPtr is better than CheckedPtr because it guarantees correct behavior, rather than just protecting against memory un-safety with a runtime crash.